package.json

```diff
@@ -1,34 +1,55 @@
 {
-  "name": "@oasisprotocol/deoxysii",
-  "version": "0.0.5",
-  "description": "Deoxys-II-256-128",
-  "main": "./deoxysii",
-  "types": "./deoxysii.d.ts",
-  "scripts": {
-    "build": "tsc",
-    "bench": "node ./bench/bench.js",
-    "test": "mocha",
-    "lint": "eslint *.js ./test ./bench"
-  },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/oasisprotocol/deoxysii-js.git"
-  },
-  "license": "MIT",
-  "author": "Oasis Protocol Foundation <info@oasisprotocol.org>",
-  "devDependencies": {
-    "@types/chai": "^4.3.4",
-    "@types/mocha": "^10.0.1",
-    "@types/node": "^18.15.11",
-    "@types/uint32": "^0.2.0",
-    "benchmark": "^2.1.4",
-    "chai": "*",
-    "eslint": "^5.14.0",
-    "mocha": "^6.1.4",
-    "typescript": "^5.0.4"
-  },
-  "dependencies": {
-    "bsaes": "0.0.2",
-    "uint32": "^0.2.1"
-  }
+	"name": "@oasisprotocol/deoxysii",
+	"version": "0.0.6",
+	"description": "Deoxys-II-256-128",
+	"sideEffects": false,
+	"files": ["dist", "!dist/*.tsbuildinfo", "src", "!src/*.json"],
+	"main": "./dist/_cjs/deoxysii.js",
+	"module": "./dist/_esm/deoxysii.js",
+	"types": "./dist/_types/deoxysii.d.ts",
+	"typings": "./dist/_types/deoxysii.d.ts",
+	"exports": {
+		"node": {
+			"import": "./dist/_esm/deoxysii.js",
+			"require": "./dist/_cjs/deoxysii.cjs",
+			"types": "./dist/_types/deoxysii.d.ts"
+		},
+		"types": "./dist/_types/deoxysii.d.ts",
+		"default": "./dist/_esm/deoxysii.js"
+	},
+	"scripts": {
+		"bench": "node ./bench/bench.js",
+		"test": "vitest",
+		"coverage": "vitest run --coverage --coverage.exclude=scripts --coverage.exclude='src/*.bench.ts'",
+		"lint": "biome check .",
+		"format": "biome format --write .",
+		"clean": "rm -rf dist",
+		"build": "npm run build:cjs && npm run build:esm && npm run build:types",
+		"build:cjs": "tsc --project ./tsconfig.build.json --module commonjs --outDir ./dist/_cjs --removeComments --verbatimModuleSyntax false && printf '{\"type\":\"commonjs\"}' > ./dist/_cjs/package.json && node scripts/rename-cjs.mjs",
+		"build:esm": "tsc --project ./tsconfig.build.json --module es2015 --outDir ./dist/_esm && printf '{\"type\": \"module\",\"sideEffects\":false}' > ./dist/_esm/package.json",
+		"build:types": "tsc --project ./tsconfig.build.json --module esnext --declarationDir ./dist/_types --emitDeclarationOnly --declaration --declarationMap",
+		"prepublishOnly": "npm run build"
+	},
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/oasisprotocol/deoxysii-js.git"
+	},
+	"license": "MIT",
+	"author": "Oasis Protocol Foundation <info@oasisprotocol.org>",
+	"devDependencies": {
+		"@biomejs/biome": "^1.7.0",
+		"@types/benchmark": "^2.1.5",
+		"@types/node": "18.x",
+		"@vitest/coverage-v8": "^1.5.3",
+		"benchmark": "^2.1.4",
+		"vitest": "^1.5.3"
+	},
+	"peerDependencies": {
+		"typescript": ">=5.0.4"
```

Comment:
This will reuse a common typescript version from one of `devDependencies`.

I should've added it to `devDependencies` though so it is not a production dependency, I have made that change.

Reply:
but if it's a dev-only dependency, then we already have full control over the dependency, so does this actually add any constraints?

Reply:
If we look at `tsup` as an example of good usage of peer dev dependencies.
`tsup`  has `5.0.2` as a `devDependency`, but they also have `typescript >=4.5.0` as a `peerDependency` with the `optional=true` flag set for typescript in `peerDependenciesMeta`. I assume 4.5.0 is the bare minimum version of typescript they want to support.

When I install `tsup` as a dev dependency their package will use whichever version of typescript I have installed. If `tsup` didn't have it as a peer dependency it's probable that I would now have two versions of typescript in the same project.

Then if I install `@oasisprotocol/deoxysii` as a devDependency of another project, with the peer dependency setup correctly, this will not potentially bring in another version of typescript, meaning typescript should act more predictably.

Is it entirely necessary... no. Do we have the same problem with the C#, OCaml, Rust, Java, PHP, C++ or C ecosystems... no (they do have their own problems, but none seem to be as asinine as the JS/Typescript/NPM ecosystem). Does Deno have the same problem... no.

Lets look at an example:

```
examples/hardhat-boilerplate/frontend$ pnpm list --depth=10 | wc -l
414739
```

Now, when people consistently use peer dependencies with typescript this makes the mess just a little less, for example:

```
examples/hardhat-boilerplate/frontend$ pnpm list --depth=10 | grep ' typescript '
│ └── typescript 4.9.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ │ │ └── typescript 5.4.5 peer
│ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ │ │ └── typescript 5.4.5 peer
│ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ │ └── typescript 5.4.5 peer
│ │ │ │ └── typescript 5.4.5 peer
│ └── typescript 5.4.5 peer
│ │ ├── typescript 5.4.5 peer
│ ├── typescript 5.4.5 peer
├── typescript 5.4.5 peer
```

Reply:
uuh, i don't know if you're using something experimental but in all established tools:
- installing @oasisprotocol/deoxysii as a devDependency does not mean that deoxysii's devDependencies will get installed
- if @oasisprotocol/deoxysii specified typescript >=4.5.0 as a dependency that would be enough to get deduplicated with existing typescript 5.4.5 in a project
- Deno should have this problem or it will have much larger problems

Reply:
Ah, sorry I've just reached my frustration point with javascript stuff.

Reply:
my understanding of peer dependencies is that they're for use at runtime. deoxysii-js doesn't use typescript at runtime. I believe we shouldn't have this entry.

tsup, when it runs, uses the typescript compiler, thus it lists typescript as a runtime dependency. it lists it in `peerDependencies` instead of `dependencies` to make sure npm doesn't end up installing a distinct version just for tsup when other tools are using a different version of typescript.